iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/helpers/IntegrationTest.java

```diff
@@ -8,9 +8,10 @@
 import org.junit.Rule;
 import org.junit.rules.TestRule;
 import org.junit.rules.TestWatcher;
+import org.junit.rules.Timeout;
 import org.junit.runner.Description;
 
-public class MethodNameLoggingIntegrationTest
+public class IntegrationTest
```

Comment:
nit: xxxxTest suggests test implementation.  But this class is used for test logging, maybe e2eTestLogging??

Reply:
This class represents our most abstract concept of an integration test, and the rules it must follow. I'm not sure e2eTestLogging covers that, especially since we now have a test timeout rule defined in this class. I'd rather keep this as IntegrationTest, or maybe change it to AbstractIntegrationTest, or something similar. For now, I think IntegrationTest works well.

Reply:
1. `abstract`
2. Name as `AbstractWatchedTest`
3. use logger
```java
protected final Logger log = LoggerFactory.getLogger(getClass());
    @Rule
    public TestRule watcher = new TestWatcher() {
        @Override
        protected void starting(Description description) {
            log.info("Test[{}] started.", description.getMethodName());
        }
        @Override
        protected void failed(Throwable e, Description description) {
            log.error("Test[{}] failed.", description.getMethodName());
        }
        @Override
        protected void succeeded(Description description) {
            log.info("Test[{}] was successful.", description.getMethodName());
        }
    };
```